lib/oban/plugins/cron.ex

```diff
@@ -107,9 +126,27 @@ defmodule Oban.Plugins.Cron do
   def handle_info(:evaluate, %State{} = state) do
     state = schedule_evaluate(state)
 
-    Query.with_xact_lock(state.conf, state.lock_key, fn ->
-      insert_jobs(state.conf, state.crontab, state.timezone)
-    end)
+    start_metadata = %{config: state.conf, plugin: __MODULE__}
+
+    :telemetry.span(
+      [:oban, :plugin],
+      start_metadata,
+      fn ->
+        state.conf
+        |> Query.with_xact_lock(state.lock_key, fn ->
+          insert_jobs(state.conf, state.crontab, state.timezone)
+        end)
+        |> case do
+          {:ok, inserted_jobs} when is_list(inserted_jobs) ->
+            metadata = %{jobs: extract_jobs_from_insert(inserted_jobs)}
```

Comment:
You can change the return value of `insert_jobs` from a list of ok tuples to a list of jobs and avoid the separate extraction step.